### PR TITLE
2193 react to subtask results settled

### LINKED ISSUE #2193: [Concent] UC4 / Additional Verification - react to `SubtaskResultsSettled`
implement the reaction to the `SubtaskResultsSettled` message as both the Provider, and as a Requestor, effectively ending the subtask's life.

* status for subtask that `SRS` was registered and the payment will be settled by the Concent

### PR DESCRIPTION
**Pull request by shadeofblue**
closes #2193

### CODE REVIEW

**Review by jiivan — APPROVED** [review]
lgtm

File: golem/transactions/incomeskeeper.py [hunk 1]
```diff
@@ -104,6 +104,18 @@ def reject(subtask_id):
             subtask_id=subtask_id
         )
 
+    @staticmethod
+    def settled(subtask_id, settled_ts):
+        try:
+            income = Income.get(subtask=subtask_id)
```
Comment (Krigpl) [inline golem/transactions/incomeskeeper.py]: I dunno what's the current state of the uniqueness of the subtask ids, but bulletproof way of grabbing the correct income is using pair `(sender, subtask_id)`, like in `update_awaiting` method.

**Review by Krigpl — APPROVED** [review]
obama_approves

File: golem/transactions/incomeskeeper.py (hunk 1 above)
Comment (shadeofblue) [reply golem/transactions/incomeskeeper.py]: @Krigpl good point indeed, sir, good point

**Pull request merged**